src/UI/state/map/actions/getFilteredData.ts

```diff
@@ -4,43 +4,74 @@ import { fetchGraphQlData } from '../../../api/api';
 import { occurrenceCsvFilterQuery } from '../../../api/queries';
 import { convertToCSV } from '../../../utils/utils';
 import { MapState, singularOutputs } from '../mapSlice';
+import { toast } from 'react-toastify';
 
 export const getFilteredData = createAsyncThunk(
   'export/getFilteredData',
   async (filters: MapState['filters']) => {
     const numberOfItemsPerResponse = 500;
-    let initTake = 0;
+    let skip = 0;
     let allData: any = [];
-    let filteredData = await fetchGraphQlData(
-      occurrenceCsvFilterQuery(
-        initTake,
-        numberOfItemsPerResponse,
-        singularOutputs(filters)
-      )
-    );
-    while (filteredData.data.OccurrenceCsvData.hasMore === true) {
-      filteredData.data.OccurrenceCsvData.items =
-        filteredData.data.OccurrenceCsvData.items.map((item: string) =>
-          JSON.parse(item)
-        );
-      allData = allData.concat(filteredData.data.OccurrenceCsvData.items);
-      initTake += numberOfItemsPerResponse;
-      filteredData = await fetchGraphQlData(
+
+    // const downloadStatus = toast.loading('fdsfsd');
+    const downloadStatus = toast.loading(' Downloading: 0%', {
+      progress: undefined,
+    });
+
+    try {
+      let filteredData = await fetchGraphQlData(
         occurrenceCsvFilterQuery(
-          initTake,
+          skip,
           numberOfItemsPerResponse,
           singularOutputs(filters)
         )
       );
+      const headers = filteredData.data.OccurrenceCsvData.items[0];
+
+      allData = filteredData.data.OccurrenceCsvData.items.slice(1);
+
+      while (filteredData.data.OccurrenceCsvData.hasMore) {
+        skip += numberOfItemsPerResponse;
+        filteredData = await fetchGraphQlData(
+          occurrenceCsvFilterQuery(
+            skip,
+            numberOfItemsPerResponse,
+            singularOutputs(filters)
+          )
+        );
+        allData = allData.concat(
+          filteredData.data.OccurrenceCsvData.items.slice(1)
+        );
+
+        toast.update(downloadStatus, {
+          render: `Downloading: ${
+            // eslint-disable-next-line prettier/prettier
+            Math.round((allData.length * 100) / filteredData.data.OccurrenceCsvData.total)
+          }%`,
+        });
+      }
+      console.log(headers, allData);
+      var file = new Blob([convertToCSV(headers, allData)], {
+        type: 'text/csv;charset=utf-8',
+      });
+
+      FileSaver.saveAs(file, 'filteredVAData.csv');
+      toast.update(downloadStatus, {
+        render: 'Download Complete',
+        type: 'success',
+        isLoading: false,
+        autoClose: 2000,
```

Comment:
You don't need the autoClose or closeOnClick things here, or below, as they're part of the defaults set in app.tsx